### PR TITLE
Add migration to create a new table for RemoteRepository and RemoteRepositoryRelation

### PR DESCRIPTION
**Pull request by saadmk11**
This PR creates a new table for the `RemoteRepository` and `RemoteRepositoryRelation` model. It also removes the fields that are not required and adds the required constraints.

ref:
https://github.com/orgs/readthedocs/projects/74#card-51212631
https://github.com/orgs/readthedocs/projects/74#card-51199356
https://github.com/orgs/readthedocs/projects/74#card-50555719

### CODE REVIEW

File: readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        migrations.CreateModel(
+            name='RemoteRepositoryRelation',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('created', django_extensions.db.fields.CreationDateTimeField(auto_now_add=True, verbose_name='created')),
+                ('modified', django_extensions.db.fields.ModificationDateTimeField(auto_now=True, verbose_name='modified')),
+                ('admin', models.BooleanField(default=False, verbose_name='Has admin privilege')),
+                ('json', jsonfield.fields.JSONField(verbose_name='Serialized API response')),
+                ('account', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='remote_repository_relations', to='socialaccount.SocialAccount', verbose_name='Connected account')),
+                ('remoterepository', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='remote_repository_relations', to='oauth.RemoteRepository')),
```
Comment (saadmk11) [inline readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py:57]: Need to rename this after https://github.com/readthedocs/readthedocs.org/pull/7771 is merged

File: readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='remote_repository_relations', to=settings.AUTH_USER_MODEL)),
+            ],
+            options={
+                'unique_together': {('remoterepository', 'account')},
```
Comment (saadmk11) [inline readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py:61]: Need to rename this after #7771 is merged

File: readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py (hunk 1 above)
Comment (humitos) [reply readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py:57]: I already merged that PR :)

File: readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                ('default_branch', models.CharField(blank=True, max_length=150, null=True, verbose_name='Default branch of the repository')),
+                ('remote_id', models.CharField(db_index=True, max_length=128)),
+                ('vcs_provider', models.CharField(choices=[('github', 'GitHub'), ('gitlab', 'GitLab'), ('bitbucket', 'Bitbucket')], max_length=32, verbose_name='VCS provider')),
+                ('organization', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='repositories', to='oauth.RemoteOrganization', verbose_name='Organization')),
+                ('project', models.OneToOneField(blank=True, null=True, on_delete=django.db.models.deletion.SET_NULL, related_name='remote_repository', to='projects.Project')),
+            ],
+            options={
+                'verbose_name_plural': 'remote repositories',
+                'db_table': 'oauth_remoterepository_2020',
+                'ordering': ['organization__name', 'name'],
```
Comment (humitos) [inline readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py:45]:
```suggestion
                'ordering': ['organization__full_name', 'full_name'],
```

Maybe `full_name` is better here?

File: readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        ),
+        migrations.AddField(
+            model_name='remoterepository',
+            name='users',
+            field=models.ManyToManyField(related_name='oauth_repositories', through='oauth.RemoteRepositoryRelation', to=settings.AUTH_USER_MODEL, verbose_name='Users'),
+        ),
+        migrations.AlterUniqueTogether(
+            name='remoterepository',
+            unique_together={('remote_id', 'vcs_provider')},
+        ),
```
Comment (humitos) [inline readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py:72]: Why this one is not in the `options` where the `RemoteRepository` model is created (some lines above)?

**Review by humitos — APPROVED** [review]
Thanks! Looks good to me!

Have you tried running this migration and do some re-sync for your connected accounts? Did you check that the old table is not touched and we have data in both?

File: readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py (hunk 4 above)
Comment (saadmk11) [reply readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py:72]: I'm not quite sure either, I auto-generated the migration with django's `manage.py makemigrations` management command.

**Comment by saadmk11** [comment]
> Have you tried running this migration and do some re-sync for your connected accounts? Did you check that the old table is not touched and we have data in both?

Yes, I have checked that! I checked out this branch and master branch one after another and made edits and re-syncs but one change does not modify the other table. Though you need to restart the celery task as well to reflect the changes. :)

File: readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py (hunk 4 above)
Comment (humitos) [reply readthedocs/oauth/migrations/0012_create_new_table_for_remote_repository_normalization.py:72]: Ah, OK. I thought you wrote this file manually. Forget about my comment, then.

**Comment by humitos** [comment]
Excellent!

**Pull request merged**